storage/blob.go

```diff
@@ -584,36 +588,62 @@ func (b BlobStorageClient) GetContainerPermissions(container string, timeout int
 	return permissionResponse, nil
 }
 
-// generateAccessPolicy generates the XML access policy used as the payload for SetContainerPermissions.
-// TODO(kpfaulkner) find better way to generate the XML. This is clunky.
-func (b BlobStorageClient) generateAccessPolicy(accessPolicy AccessPolicyDetails) string {
+func (b BlobStorageClient) generatePermissions(accessPolicy AccessPolicyDetails) (permissions string) {
+	// generate the permissions string (rwd).
+	// still want the end user API to have bool flags.
+	permissions = ""
 
-	if accessPolicy.ID != "" {
-		s := `<?xml version="1.0" encoding="utf-8"?><SignedIdentifiers><SignedIdentifier>`
-		s += fmt.Sprintf("<Id>%s</Id>", accessPolicy.ID)
-		s += "<AccessPolicy>"
+	if accessPolicy.CanRead {
+		permissions += "r"
+	}
 
-		permissions := ""
-		if accessPolicy.CanRead {
-			permissions += "r"
-		}
+	if accessPolicy.CanWrite {
+		permissions += "w"
+	}
 
-		if accessPolicy.CanWrite {
-			permissions += "w"
-		}
+	if accessPolicy.CanDelete {
+		permissions += "d"
+	}
 
-		if accessPolicy.CanDelete {
-			permissions += "d"
+	return permissions
+}
+
+// convertAccessPolicyToXMLStructs converts between AccessPolicyDetails which is a struct better for API usage to the
+// AccessPolicy struct which will get converted to XML.
+func (b BlobStorageClient) convertAccessPolicyToXMLStructs(accessPolicy AccessPolicyDetails) (accessPolicyToMarshal AccessPolicy) {
+	accessPolicyToMarshal = AccessPolicy{}
+	accessPolicyToMarshal.SignedIdentifiersList = SignedIdentifiers{}
+	accessPolicyToMarshal.SignedIdentifiersList.SignedIdentifiers = []SignedIdentifier{}
+	si := SignedIdentifier{}
+	si.ID = accessPolicy.ID
+	si.AccessPolicy = AccessPolicyDetailsXML{}
+	si.AccessPolicy.StartTime = accessPolicy.StartTime
+	si.AccessPolicy.ExpiryTime = accessPolicy.ExpiryTime
+	permissions := b.generatePermissions(accessPolicy)
+	si.AccessPolicy.Permission = permissions
+	accessPolicyToMarshal.SignedIdentifiersList.SignedIdentifiers = append(accessPolicyToMarshal.SignedIdentifiersList.SignedIdentifiers, si)
+	return accessPolicyToMarshal
+}
+
+// generateAccessPolicy generates the XML access policy used as the payload for SetContainerPermissions.
+func (b BlobStorageClient) generateAccessPolicy(accessPolicy AccessPolicyDetails) (accessPolicyXML string, err error) {
+
+	if accessPolicy.ID != "" {
+		ap := b.convertAccessPolicyToXMLStructs(accessPolicy)
+		body, _, err := xmlMarshal(ap.SignedIdentifiersList)
```

Comment:
ok, have done 2 things to "resolve" it.
1)  make sure the tests round their timestamps down to the second.
2) make sure the XML generation code rounds it as well (yes, strictly dont need #1 then, but it's there already so thought I'd leave it). 

Am wondering if the number of decimal places is based off platform? I'm coding and testing on Windows where I assume travis is on Linux?